__tests__/app.test.js

```diff
@@ -1,20 +1,41 @@
-const db = require("../db/connection")
+const db = require("../db/connection");
 const request = require("supertest");
 const app = require("../app");
-const seed = require("../db/seeds/seed")
-const data = require("../db/data/test-data/index")
+const seed = require("../db/seeds/seed");
+const data = require("../db/data/test-data/index");
 
 beforeEach(() => {
-    return seed(data);
-})
+  return seed(data);
+});
 
 afterAll(() => {
-    return db.end();
-})
+  return db.end();
+});
 
 describe("GET /api/healthcheck ", () => {
-    test("STATUS -200 respons with a status of 200", () => {
-        return request(app).get("/api/healthcheck")
-        .expect(200);
+  test("STATUS -200 respons with a status of 200", () => {
+    return request(app).get("/api/healthcheck").expect(200);
+  });
+});
+
+describe("GET", () => {
+  describe("GET /api/topics", () => {
+    test("STATUS - 200 responds with a status of 200 and array of objects", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then((response) => {
+          expect(response.body.topics).toHaveLength(3);
+          response.body.topics.forEach((topic) => {
+            expect(topic).toHaveProperty("slug");
+            expect(topic).toHaveProperty("description");
+          });
+        });
+    });
+    test("STATUS - 400 respond with a message bad request when people request with an invalid endpoint", () => {
+        return request(app)
+        .get("/api/topicafjajfh")
+        .expect(404);
```

Comment:
A couple of points on this one:

- Your test description isn't accurate here. This is a 404 test and currently isn't returning a message
- Express has a built in feature that will automatically respond with a 404 status and no message when an invalid endpoint is used. Therefore, this test in its current form isn't really needed as the devs who built Express will have already tested that feature. You could build this up a bit more and have it respond with a message as well 😄

This isn't urgent and you don't have to do it right now. However, I do highly recommend that you do at some point 🙂

Reply:
Thanks for the feedback Ray, 

I think I forgot to give it one final check before I rushed to submitting. My test description is horrible! I'm expecting a status 400 but expecting 404!?!! Sorry I didn't check. I was playing a bit with it and didn't change it.  I will fix it at some point today definitely :)